src/main/java/com/avispl/symphony/dal/avdevices/encoderdecoder/haivision/xdecoder/common/decoder/controllingmetric/DecoderControllingMetric.java

```
@@ -0,0 +1,64 @@
+/*
+ * Copyright (c) 2022 AVI-SPL, Inc. All Rights Reserved.
+ */
+package com.avispl.symphony.dal.avdevices.encoderdecoder.haivision.xdecoder.common.decoder.controllingmetric;
+
+import java.util.Arrays;
+import java.util.Optional;
+
+/**
+ * Set of decoder controlling metric keys
+ *
+ * @author Harry / Symphony Dev Team<br>
+ * Created on 5/3/2022
+ * @since 1.0.0
+ */
+public enum DecoderControllingMetric {
+
+	PRIMARY_STREAM("PrimaryStream"),
+	SECONDARY_STREAM("SecondaryStream"),
+	STILL_IMAGE("StillImage"),
+	STILL_IMAGE_DELAY("StillImageDelay"),
+	SYNC_MODE("EnableBuffering"),
+	BUFFERING_DELAY("Delay"),
+	BUFFERING_MODE("BufferingMode"),
+	MULTI_SYNC_BUFFERING_DELAY("MultiSyncDelay"),
+	OUTPUT_RESOLUTION("OutputFrameRate"),
+	OUTPUT_FRAME_RATE("OutputFrameRate"),
+	STATE("Active"),
+	APPLY_CHANGE("ApplyChange"),
+	CANCEL("Cancel"),
+	EDITED("Edited");
+
+	private final String name;
+
+	/**
+	 * Parameterized constructor
+	 *
+	 * @param name Name of decoder monitoring metric
+	 */
+	DecoderControllingMetric(String name) {
+		this.name = name;
+	}
+
+	/**
+	 * retrieve {@code {@link #name}}
+	 *
+	 * @return value of {@link #name}
+	 */
+	public String getName() {
+		return this.name;
+	}
+
+	/**
+	 * This method is used to get decoder controlling metric by name
+	 *
+	 * @param name is the name of decoder controlling metric that want to get
+	 * @return DecoderControllingMetric is the decoder controlling metric that want to get
+	 */
+	public static DecoderControllingMetric getByName(String name) {
+		Optional<DecoderControllingMetric> decoderControllingMetric = Arrays.stream(DecoderControllingMetric.values()).filter(com -> com.getName().equals(name)).findFirst();
+		return decoderControllingMetric.orElse(null);
```

Throw exception here if fail to get DecoderControllingMetric by name, instead of return null.
For example:
throw new IllegalArgumentException("Can not find the enum with name: " + name);

---

I fixed it